.github/workflows/build.yml

```diff
@@ -57,35 +64,59 @@ jobs:
         name: mac-binary
         path: build/asciidoctor-web-pdf-mac.zip
       if: matrix.primary
-    - name: Smoke test
-      run: |
-        unzip asciidoctor-web-pdf-linux.zip -d out
-        ./out/asciidoctor-web-pdf ./out/examples/document/basic-example.adoc -o basic.pdf
-        ls basic.pdf
-      working-directory: build
+    - name: Upload Linux artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: linux-binary
+        path: build/asciidoctor-web-pdf-linux.zip
       if: matrix.primary
+  test-linux:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: linux-binary
+      - name: unzip
+        run: |
+          unzip asciidoctor-web-pdf-linux.zip -d out
+      - name: Smoke test
+        run: |
+          cd out/asciidoctor-web-pdf-linux-${{ needs.build.outputs.version }}
+          rm -rf basic.pdf
```

Comment:
We don't need to remove this file since it does not exist. Also the `-r` seems a bit excessive to remove a single file.

Reply:
Removal because such a file could be added by accident as part of the distribution. While all seems to be excessive, it  really asserts that asciidoctor-web-pdf generated that file. The -r is necessary cause it basic.pdf could also be a folder..